### PR TITLE
SDIT-900 Improve PUT /api/bookings/activities/attendance

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 1]
```diff
@@ -325,36 +326,43 @@ public List<ScheduledEvent> getBookingActivities(final Long bookingId, final Loc
...
-            throw EntityNotFoundException.withMessage("Activity Id %d not found", activityId);
-        }
+            updateAttendance(activityId, updateAttendance, bookingMap.get(bookingActivity.getBookingId()), activityOutcome);
+        });
+    }
+
+    private void updateAttendance(final Long activityId,
+                                  final UpdateAttendance updateAttendance,
+                                  final OffenderSummary offenderSummary, final PayableAttendanceOutcomeDto activityOutcome) {
+        bookingRepository.updateAttendance(offenderSummary.getBookingId(), activityId, updateAttendance, activityOutcome.isPaid(), activityOutcome.isAuthorisedAbsence());
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:365]: essentially pulled out the selects within the loop

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ public void testThatUpdateAttendanceIsLockedDown_WhenPayRoleIsMissing() {
             },
             -2, -11);
 
-        assertThat(response.getStatusCodeValue()).isEqualTo(403);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java:142]: getStatusCodeValue is deprecated

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 3]
```diff
@@ -325,36 +326,35 @@ public List<ScheduledEvent> getBookingActivities(final Long bookingId, final Loc
...
     }
 
     @Transactional
     @PreAuthorize("hasRole('ROLE_PAY')")
     public void updateAttendanceForMultipleBookingIds(final Set<BookingActivity> bookingActivities, @Valid @AttendanceTypesValid final UpdateAttendance updateAttendance) {
-        bookingActivities.forEach(bookingActivity -> updateAttendance(bookingActivity.getActivityId(), updateAttendance, getLatestBookingByBookingId(bookingActivity.getBookingId())));
-    }
-
-    private void updateAttendance(final Long activityId, final UpdateAttendance updateAttendance, final OffenderSummary offenderSummary) {
-        validateActivity(activityId);
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:344]: It turned out that both calling getLatestBookingByBookingId(), and validateActivity(), were redundant as the same checks are done in the update itself

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**